src/wl_window.c

```diff
@@ -2584,23 +2584,21 @@ const char* _glfwGetScancodeNameWayland(int scancode)
         return NULL;
     }
 
-    const uint32_t codepoint = _glfwKeySym2Unicode(keysyms[0]);
-    if (codepoint == GLFW_INVALID_CODEPOINT)
+    // HACK: xkb_keysym_to_utf8() requires the third parameter (size of the output buffer) 
+    // to be at least 7 (6 bytes + a null terminator), because it was written when UTF-8
+    // sequences could be up to 6 bytes long. The _glfw.wl.keynames buffers are only 5 bytes
+    // long, because UTF-8 sequences are now limited to 4 bytes and no codepoints were ever assigned
+    // that needed more than that. To work around this, we lie to the function about the buffer
+    // size, because we know it won't use more than 5 bytes.
+    //
+    // See: https://github.com/xkbcommon/libxkbcommon/issues/418
+    if (xkb_keysym_to_utf8(keysyms[0], _glfw.wl.keynames[key], 7) <= 0)
```

Comment:
I'd especially like comments on if this is okay.

Reply:
I think we should use a temp buffer here, and if the returned number of bytes written is more than the size of `keynames[key]` we fail otherwise copy into `keynames`.

The cost of a copy should be extremely small in this circumstance, and we'd rather have safer code.

Reply:
Note that I can make the above change if needed, but if you have time to do so please go ahead and push to your branch.

Reply:
I've made that change.